### DISCUSSION TITLE
Missing function about DoReFa-Quantizer

### DISCUSSION THREAD

**Discussion by loveklmn** [microsoft/nni#4213] [Q&A] [answered]
The paper about DoReFa-Net present 3 quantization steps ( gradient, weight, and  [activation || output ]). Seems that only `weight` was implemented. Is it a mistake? Do `nni` research group have any plan to do about it?
https://github.com/microsoft/nni/blob/master/docs/en_US/Compression/Quantizer.rst

**Comment by linbinskn** [comment, marked as answer]
Exactly, weight & output & gradient are all quantized in paper. And both `weight` and `output` quantization have been supported in NNI's `DoReFa-Net` currently. We think the priority of implementing gradient quantization is low since `DoReFa-Net` is not very suitable for deployment when comparing to another training aware quantization algorithm such as QAT.
